### PR TITLE
Converted Hardcoded Whitelist to Env Var

### LINKED ISSUE #17: Change hard-coded whitelisted orgs to environment variables
Org whitelist should be dynamically set by environment variables, likely semi-colon separated. `maryvilledev;LaunchCodeEducation`.
Additionally, export in prod to allow MT and LC, and dev MT only.

### PR DESCRIPTION
**Pull request by Hopding**
## Description

This PR makes it necessary to specify a `ORG_WHITELIST` environment variable containing a semicolon-separated list of whitelisted GitHub organization for the `GitHubAccessCodeGetter` lambda. e.g. `ORG_WHITELIST = "maryvilledev;LaunchCodeEducation"`. Formerly, the whitelist was hard coded into the lambda itself.

## Motivation and Context

This change allows us to easily update the organization whitelist.

Fixes #17.

### CODE REVIEW

File: publish [hunk 1]
```diff
@@ -24,7 +24,7 @@ function nodeInstall {
 }
 
 function nodeCleanup {
-  printf "Removing node_modules and etc for $1 lambda"
+  printf "Removing node_modules and etc for $1 lambda\n"
```
Comment (dane-johnson) [inline publish]: `printf "\n"` == `echo`

File: lambdas/GitHubAccessCodeGetter/index.js [hunk 2]
```diff
@@ -3,12 +3,39 @@
...
+}
+const orgUrl = `https://api.github.com/user/orgs`
+const orgOpts = (token) => ({
+  headers: {
+    'Authorization': `token ${token}`
+  }
+})
+
+// Create whitelist array, after checking for the env var
+if (!process.env.Whitelist) console.err('Whitelist env var missing!')
```
Comment (dane-johnson) [inline lambdas/GitHubAccessCodeGetter/index.js]: Environment variables should be matchstick-seperated and all-caps i.e. `ORG_WHITELIST`

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,4 @@
 .env
-node_modules
+**/node_modules/**
```
Comment (dane-johnson) [inline .gitignore]: Can I get an explanation for all this?

Comment (Hopding) [inline .gitignore]: Using `**` as a prefix and postfix to the directory name causes it to be ignored regardless of its location in the file tree.
https://git-scm.com/docs/gitignore explains it more in depth.

File: lambdas/GitHubAccessCodeGetter/index.js (hunk 2 above)
Comment (Hopding) [inline lambdas/GitHubAccessCodeGetter/index.js]: Good point, will fix. This is actually a problem with some of our other Lambda's env vars, they don't all adhere to the same naming convention.

File: publish (hunk 1 above)
Comment (Hopding) [inline publish]: I prefer to use `printf` over `echo` in bash scripts, I have found it to be more intuitive and consistent. http://unix.stackexchange.com/questions/65803/why-is-printf-better-than-echo#65819 explains their differences pretty in depth.

**Comment by thebho** [comment]
Added whitelist variable to env

**Pull request merged**